src/main/java/com/google/api/codegen/transformer/ApiMethodTransformer.java

```diff
@@ -201,6 +441,16 @@ private RequestObjectParamView generateRequestObjectParam(
     return allDocs;
   }
 
+  public SimpleParamDocView getRequestObjectParamDoc(
```

Comment:
I tried it out and I don't think it simplifies anything.

* setRequestObjectMethodFields: only needs to wrap the result of getRequestObjectParamDoc in Arrays.asList()
* generateOptionalArrayMethodDoc: just adds the result to an existing List that it is building up.

For both of these cases, they logically create a singleton doc. I don't think it makes sense for them to return a List.